### ISSUE TITLE
Fix S1226: Variables that are passed as an argument to a method should be considered as read

### ISSUE THREAD

**Issue by valhristov** [SonarSource/sonar-dotnet#2249]
The following code snippet should not raise an issue:

```
...
catch (Exception e)
{
    while (e != null)
    {
        list.Add(FormatMessage(e));
        e = e.InnerException;
    }
}
```

_Originally posted by @rymeskar in https://github.com/SonarSource/sonar-dotnet/issues/630#issuecomment-457177166_

Tip: the rule considers a variable read only if it appears on the right hand side of an assignment.

**Comment by andrei-epure-sonarsource** [comment]
Could not reproduce, I added reproducing tests in #2372
